app/controllers/spaces.rb

```diff
@@ -0,0 +1,64 @@
+class MakersBnb < Sinatra::Base
+
+  get '/edit/:space_id' do
+    unless current_user
+      redirect '/sessions/new'
+    end
+    @space = Space.get(params[:space_id])
+    if current_user.id == @space.user_id
+      erb :'spaces/edit'
+    else
+      redirect '/users/account'
+    end
+  end
+
+  post '/edit'do
+    Space.get(params[:space_id]).update(name: params[:name], description: params[:description], price: params[:price])
+    redirect to 'users/account'
+  end
+
+  get '/spaces' do
+    if session[:space_array]
+      @spaces = session[:space_array].map do |space_id|
+        Space.get(space_id)
+      end
+    else
+      @spaces = Space.all
+    end
+    session[:space_array] = nil
+    erb :'spaces/index'
+  end
+
+  post '/spaces' do
+    validate_dates(params[:available_from], params[:available_to], '/spaces')
+    find_available_spaces(params[:available_from], params[:available_to])
```

Comment:
And this I like! Although I would prefer these to be pushed into proper classes, not just helpers. That is, a 'service object' - which looks like this:

```ruby
FindAvailableSpaces.with(params[:available_from], params[:available_to])
```

Bear in mind that here, though, you really shouldn't be using a POST action to filter `Spaces`. This doesn't change anything in the database, so you should be GETting them instead.

```ruby
get '/spaces' do
  dates = ValidateDates.with(params[:available_from], params[:available_to])
  @spaces = Space.filter_by_dates(dates)
  erb :'spaces/index'
end
```